src/controllers/sendMessage.js

```diff
@@ -1,13 +1,27 @@
-import http from "http";
+import http from "http"
+import saveMessage from "../clients/saveMessage.js"
+import checkBudget from "../clients/checkBudget.js"
+import saveAmount from "../clients/saveAmount.js"
 
-import saveMessage from "../clients/saveMessage.js";
+
+
+const MESSAGE_PRICE = 2
 
 export default async (req, res) => {
-  const body = JSON.stringify(req.body);
+
+  const body = JSON.stringify(req.body)
+
+  const actualMoney = await checkBudget()
+
+  if (actualMoney.amount < MESSAGE_PRICE) {
+    return res.status(500).json("Please Top Up")
+  } else {
+    saveAmount(-MESSAGE_PRICE)
```

Comment:
Be careful with the naming, saveAmount doesn't save a new amount, it changes the credit by X amount. A better name could be changeBudgetBy(-2)